src/v/pandaproxy/schema_registry/handlers.cc

```diff
@@ -321,41 +319,6 @@ get_schemas_ids_id(server::request_t rq, server::reply_t rp) {
     co_return rp;
 }
 
-struct schema_version_json : public ss::json::json_base {
```

Comment:
If I understand it correctly, the motivation for reverting this is to simplify the code, using the fact that there’s now no risk of oversized allocations with `chunked_buffer`.

* The streaming version also avoids a temporary copy of the whole serialized state in memory. Is it a problem that we now have this extra temporary again?
* Are the CPU cache effects (mainly cache pollution but also reading less-recently-written data) of having a temporary copy of the whole serialized state a problem?
* Would it be difficult to have a streaming output writer version of `rjson_serialize` similar to `stream_range_as_array` that takes a container of items, serializes them and writes them out one at a time?
* Can the serde of large objects be expensive enough to cause reactor stalls?

Reply:
1. The streaming version copies every object, which usually contains strings, and therefore performs many allocations
2. I imagine the CPU caching is better with fewer larger contiguous block copies than it is with many small ones
3. I plan to make an async `rjson_serialize` at some point in the future
4. Possibly